src/main/helm/helm-chart-manager.ts

```diff
@@ -20,86 +20,100 @@
       logger.error("HELM-CHART-MANAGER]: failed to list charts", { error });
 
       return {};
     }
   }
 
-  public async getReadme(name: string, version = "") {
+  private async command(action: string, name: string, version?: string) {
     const helm = await helmCli.binaryPath();
+    const cmd = [`"${helm}" ${action} ${this.repo.name}/${name}`];
 
-    if(version && version != "") {
-      const { stdout } = await promiseExec(`"${helm}" show readme ${this.repo.name}/${name} --version ${version}`).catch((error) => { throw(error.stderr);});
+    if (version) {
+      cmd.push("--version", version);
+    }
 
-      return stdout;
-    } else {
-      const { stdout } = await promiseExec(`"${helm}" show readme ${this.repo.name}/${name}`).catch((error) => { throw(error.stderr);});
+    try {
+      const { stdout } = await promiseExec(cmd.join(" "));
 
       return stdout;
+    } catch (error) {
+      throw error.stderr || error;
     }
   }
 
-  public async getValues(name: string, version = "") {
-    const helm = await helmCli.binaryPath();
-
-    if(version && version != "") {
-      const { stdout } = await promiseExec(`"${helm}" show values ${this.repo.name}/${name} --version ${version}`).catch((error) => { throw(error.stderr);});
-
-      return stdout;
-    } else {
-      const { stdout } = await promiseExec(`"${helm}" show values ${this.repo.name}/${name}`).catch((error) => { throw(error.stderr);});
+  public async getReadme(name: string, version?: string) {
+    return this.command("show readme", name, version);
+  }
 
-      return stdout;
-    }
+  public async getValues(name: string, version?: string) {
+    return this.command("show values", name, version);
   }
 
-  protected async cachedYaml(): Promise<CachedYaml> {
-    if (!(this.repo.name in this.cache)) {
+  protected async cachedYaml(): Promise<RepoHelmChartList> {
+    if (!HelmChartManager.#cache.has(this.repo.name)) {
       const cacheFile = await fs.promises.readFile(this.repo.cacheFilePath, "utf-8");
-      const data = yaml.safeLoad(cacheFile);
-
-      for(const key in data["entries"]) {
-        data["entries"][key].forEach((version: any) => {
-          version["repo"] = this.repo.name;
-          version["created"] = Date.parse(version.created).toString();
-        });
+      const { entries } = yaml.safeLoad(cacheFile) as { entries: RepoHelmChartList };
+
+      /**
+       * Do some initial preprocessing on the data, so as to avoid needing to do it later
+       * 1. Set the repo name
+       * 2. Normalize the created date
+       * 3. Filter out deprecated items
+       */
+
+      const normalized = Object.fromEntries(
+        Object.entries(entries)
+          .map(([name, charts]) => [
+            name,
+            sortCharts(
+              charts.map(chart => ({
+                ...chart,
+                created: Date.parse(chart.created).toString(),
+                repo: this.repo.name,
+              })),
+              logger.warn,
```

Comment:
Just to make sure. Will `logger.warm` be okay for both prod and dev environment?

Reply:
Yes of course, though this was mostly don't so that we don't have warnings in the tests, I think I will change this.